### PR TITLE
feat(gorilla/mux): adds support for gorilla/mux.

### PR DESCRIPTION
**Pull request by jcchavezs**
This PR adds support for `gorilla/mux`.

TODO: verify body/headers and being recorded as expected.

### CODE REVIEW

File: instrumentation/opentelemetry/github.com/gorilla/hypermux/mux.go [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+	spanName, err = route.GetPathTemplate()
+	if err != nil {
+		spanName, err = route.GetPathRegexp()
+	}
+
+	return
+}
+
+// NewMiddleware sets up a handler to start tracing the incoming requests.
+func NewMiddleware() mux.MiddlewareFunc {
```
Comment (jcchavezs) [inline instrumentation/opentelemetry/github.com/gorilla/hypermux/mux.go:29]: This is the actual Gorilla mux instrumentation.

File: instrumentation/opentelemetry/github.com/gorilla/hypermux/mux_test.go [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+	res, err := http.DefaultClient.Do(req)
+	if err != nil {
+		t.Errorf("unexpected error: %v", err)
+	}
+
+	if want, have := 202, res.StatusCode; want != have {
+		t.Errorf("unexpected status code, want %q, have %q", want, have)
+	}
+
+	spans := flusher()
```
Comment (jcchavezs) [inline instrumentation/opentelemetry/github.com/gorilla/hypermux/mux_test.go:58]: We also need to verify body/headers are being recorded.

File: instrumentation/opentelemetry/github.com/gorilla/hypermux/mux.go [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+)
+
+func spanNameFormatter(operation string, r *http.Request) (spanName string) {
+	route := mux.CurrentRoute(r)
+	if route == nil {
+		return
+	}
+
+	var err error
+	spanName, err = route.GetPathTemplate()
```
Comment (tim-mwangi) [inline instrumentation/opentelemetry/github.com/gorilla/hypermux/mux.go:20]: spanName is always nil when err is not? In that case it will be modified, no?

Comment (jcchavezs) [reply instrumentation/opentelemetry/github.com/gorilla/hypermux/mux.go:20]: if `err != nil`, spanName is `""`, still we need it as types must be respected.

Comment (tim-mwangi) [reply instrumentation/opentelemetry/github.com/gorilla/hypermux/mux.go:20]: In that case to we want an empty spanName? Shouldn't we return the value that was passed in to this method?

Comment (jcchavezs) [reply instrumentation/opentelemetry/github.com/gorilla/hypermux/mux.go]: Good point. I will default to `request.Method`.

**Review by tim-mwangi — APPROVED** [review]

File: instrumentation/opentelemetry/github.com/gorilla/hypermux/mux.go (hunk 3 above)
Comment (tim-mwangi) [reply instrumentation/opentelemetry/github.com/gorilla/hypermux/mux.go]: ok.

**Pull request merged**